Importer/Interpreter/RelationshipDataInterpreter.php

```diff
@@ -29,66 +34,102 @@ class RelationshipDataInterpreter implements InterpreterInterface
     private $lookupFieldInterpreter;
 
     /**
+     * @var LogInterface
+     */
+    private $log;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
      * @param RelationshipConfigurationInterface $configuration
      * @param EntityManager                      $entityManager
+     * @param EventDispatcherInterface           $eventDispatcher
+     * @param LogInterface                       $log
      */
-    public function __construct(RelationshipConfigurationInterface $configuration, EntityManager $entityManager)
-    {
+    public function __construct(
+        RelationshipConfigurationInterface $configuration,
+        EntityManager $entityManager,
+        EventDispatcherInterface $eventDispatcher,
+        LogInterface $log
+    ) {
         $this->configuration = $configuration;
         $this->lookupFieldInterpreter = new LookupFieldInterpreter($entityManager);
         $this->entityManager = $entityManager;
+        $this->log = $log;
+        $this->eventDispatcher = $eventDispatcher;
     }
 
     /**
      * @param array $data
+     * @param int   $maxErrorAmount
      * @param bool  $associative
      *
-     * @return null
+     * @return array
      */
-    public function interpret(array $data, $associative = true)
+    public function interpret(array $data, $maxErrorAmount, $associative = true)
```

Comment:
FYI: I tested the import of relationships which works in `u2` with some small changes here and there. Here are some problems I had, that we have to solve if we decide to add the relationship import to `u2`.

**Scenario 1: Importing a unit to a master file that already exists**

On the csvImporter, when we flush, a DBALException is thrown, that duplicated record was found. Everything worked well, and a configuration error was added to the log. The problem was that on this exception, the entity manager closes. So after calculating the import statistics, we update the status of the import to 'completed' and we try to flush again. 

So this fails because the entity manager is closed.

You can check `src/CoreBundle/ProcessSession/ProcessSessionManager.php:111`

**Scenario 2: Importing relationships for Master file that belong to a closed period.**

Here we also do not validate the updated entity so we can create new relationships for objects that belong to a closed period. 

You can check how we achieve validation in entity data interpreter:
- vendor/netdudes/importer-bundle/Netdudes/ImporterBundle/Importer/Interpreter/EntityDataInterpreter.php:157
 
**Scenario 3: Importing relationships for Master file**
In our logic after we import some data we flush, and then we change the status of the import to `completed` and we flush again.

I was not able to find out why, but after creating for example a new relationship between master file and unit, when I flushed in order to change the status of the import, the same relationship was again in the unit of work, so I got a duplicated record exception.

I tried to flush only the `import` but still I had the same problem. I also tried to detach updated entity after flushing the first time but nothing. Probably I am missing something here.